### PR TITLE
feat: add support for oidc authentication

### LINKED ISSUE #19741: Camunda can be configured to use an OIDC Identity Provider
### Description

Camunda supports OIDC authentication via the profile `auth-oidc`.

Where possible we should support existing configuration options documented in https://docs.camunda.io/docs/next/self-managed/setup/guides/connect-to-an-oidc-provider/?optionsType=env.

```[tasklist]
### Breakdown
- [ ] Authentication with an OIDC provider can be configured via application properties and env variables
```

### PR DESCRIPTION
**Pull request by maryarm**
## Description

Camunda supports OIDC authentication via the profile auth-oidc.

Where possible we should support existing configuration options documented in https://docs.camunda.io/docs/next/self-managed/setup/guides/connect-to-an-oidc-provider/?optionsType=env.

## Related issues

closes #19741

### CODE REVIEW

File: dist/src/main/resources/application-auth-oidc.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+spring:
```
Comment (Ben-Sheppard) [inline dist/src/main/resources/application-auth-oidc.yaml:1]: 🔧 I think that the typical place for these values is in `application-<profile>.properties` - I don't think there is anything functionally different between the two but I think its good to use a single approach here if possible.

File: dist/src/main/resources/application-auth-oidc.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    oauth2:
+      client:
+        registration:
+          oidcclient:
+            client-id:
+            client-secret:
+            authorization-grant-type: authorization_code
+            redirect-uri:
+            provider: oidcclient
+            scope:
```
Comment (Ben-Sheppard) [inline dist/src/main/resources/application-auth-oidc.yaml:12]: 💭 Something for the future but we've had instance in the past where customers have wanted configurable scopes, I guess we can add this in later though if necessary

**Review by Ben-Sheppard — APPROVED** [review]
LGTM - I can't really see any tests that we can add here that bring value or that aren't just testing the sub-systems

File: dist/src/main/resources/application-auth-oidc.yaml (hunk 1 above)
Comment (maryarm) [reply dist/src/main/resources/application-auth-oidc.yaml:1]: Do you mean to switch from yaml to properties? I find it more readable in cases like this that there is a multi level hierarchy.

File: dist/src/main/resources/application-auth-oidc.yaml (hunk 2 above)
Comment (maryarm) [reply dist/src/main/resources/application-auth-oidc.yaml:12]: I see. Doesn't the default way of property overriding for this case help? for example customer can override the scope property here with their new list of profile.

Comment (Ben-Sheppard) [reply dist/src/main/resources/application-auth-oidc.yaml:12]: Good point, it does, but i'm not sure if we want to make it a nicer experience down the line i.e. an env variable. Not critical for now at all but wanted to mention it

File: dist/src/main/resources/application-auth-oidc.yaml (hunk 1 above)
Comment (Ben-Sheppard) [reply dist/src/main/resources/application-auth-oidc.yaml:1]: Yea so elsewhere in the code we have things like `application-identity.properties` for all the identity related things. NABD but I wanted to raise that it goes against the current approach

Comment (maryarm) [reply dist/src/main/resources/application-auth-oidc.yaml:1]: I'll apply your comment in my next PR (loading claims in authentication object when login with oidc)

**Pull request merged**